server-protocol.md

```diff
@@ -92,6 +85,33 @@ following keys (and ignores all others):
   future version of the protocol requires a rate-limiting CAPTCHA ticket or
   other authorization record, the server can send `error` (explaining the
   requirement) if it does not see this ticket arrive before the `bind`.
+* `permission-required`: a set of available authentication methods,
+  proof of work challenges etc. The client needs to "solve" one of
+  them in order to get access to the service.
+
+The client should examine the `permissions-required` methods (if
+present) and select one to use (see also "Permission to Use the
+Server" below).
+
+* If the client doesn't send a `submit-permissions` message (or it is
+  invalid or otherwise unacceptable) the server either proceeds anyway
+  (because it is allowing all access) or sends a "permission denied"
+  error message and closes the connection.
+
+* If the client doesn't support any permissions methods it should show
+  an error to the user and disconnect. (If the server is allowing
+  permission-less connections it should include the method `none` in
+  its list).
+
+* Backwards compatibility: older clients will always send the `bind`
+  message immediately (without waiting for the `welcome`
+  message). Server implementations should proceed as with step 1.
```

Comment:
There is no "step 1" anymore in this version :)

Reply:
re-worded this